### PR TITLE
Add Banned Tokens support

### PR DESCRIPTION
**Pull request by ihatenumbers**
Add support for banned tokens, or anti-slop.

### CODE REVIEW

**Comment by neCo2** [comment]
Can anti-slop only ban strings, or also *bias* them?

In any case, `bannedTokens` etc. should probably be renamed to `bannedStrings` to avoid confusion, since token banning is already a thing with logit bias.

Definitely wait for the maintainer's feedback on this, but you could also hijack the logit bias modal for this to save on UI space.
You can just intercept any multi-token strings in or before the `SetLogitBiasParams` functions, and if it's banned (bias power = -100), add them to `bannedTokens`.

**Comment by ihatenumbers** [comment]
I named it `bannedTokens` since the API named `banned_tokens` as parameter.

In KoboldAI Lite documentation:
`banned_tokens` - _An array of string sequences, each entry represents a word or phrase prevented from being generated, either by modifying the model vocab or by backtracking and regenerating when they appear._

I don't think there's any relation to logit bias. Also, I don't think phrases would work as logit bias.

**Comment by neCo2** [comment]
> I named it bannedTokens since the API named banned_tokens as parameter.

Oh, I see. Makes sense to name it that way then.

> Also, bias power -100 would be different from banning since it still permits the token during generation

A -100 in mikupad completely bans the token (in llamaCPP at least. I believe I couldn't find an equivalent for the other two APIs. Been a while since I wrote it.)
Here's the relevant line:
https://github.com/lmg-anon/mikupad/blob/cc7a16c8b8b1d462f25eb3efe4973c47b2b7a153/mikupad.html#L3197

But yeah, if it's a completely seperate thing from bias, maybe it should stay as a separate UI element. I just think that a small field with an array, while functional, isn't great to use, but that's nitpicking.

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:6413]:
```suggestion
					<${Checkbox} label="Banned Strings"
```

File: mikupad.html
Comment (lmg-anon) [inline mikupad.html:6531]:
```suggestion
								<${InputBox} label="Banned Strings (JSON array)" type="text" pattern="^\\[.*?\\]$"
```

**Review by lmg-anon — CHANGES_REQUESTED** [review]

**Comment by lmg-anon** [comment]
This should do it, thank you for the contribution!

**Pull request merged**
